.github/workflows/ci.yml

```diff
@@ -163,44 +168,86 @@ jobs:
       - name: Validation tests
         run: pytest tests/validation --no-cov --verbose
 
-  publish:
+  github-release:
     needs:
       - build-python
       - unit-tests
       - validation-tests
       - sonar-cloud
     permissions:
       contents: write
-    env:
-      TWINE_USERNAME: ${{ secrets.PYPI_USER }}
-      TWINE_PASSWORD: ${{ secrets.PYPI_PASS }}
     runs-on: ubuntu-latest
     steps:
-      - name: Setup Python 3.11
+      - name: Setup Python 3.13
         uses: actions/setup-python@v5
         with:
-          python-version: "3.11"
+          python-version: "3.13"
 
       - name: Load built wheel file
         uses: actions/download-artifact@v4
         with:
           name: power-grid-model-io
           path: wheelhouse/
 
-      - name: Upload wheels
-        if: (github.event_name == 'push') || ((github.event_name == 'workflow_dispatch') && (github.event.inputs.create_release == 'true'))
-        run: |
-          pip install twine
-          echo "Publish to PyPI..."
-          twine upload --verbose wheelhouse/*
+      - name: Get tag
+        id: tag
+        run: echo "tag=v${{ needs.build-python.outputs.version }}" >> $GITHUB_OUTPUT
```

Comment:
any particular reason you use the `needs.build-python.output.version` instead of the `PYPI_VERSION` directly?